grails-app/services/de/iteratec/osm/measurement/environment/wptserver/ResultPersisterService.groovy

```diff
@@ -455,6 +456,15 @@ class ResultPersisterService implements iResultListener {
         return !viewTag.getProperty(tag).isEmpty() && viewTag.getProperty(tag).toString().isInteger() && viewTag.getProperty(tag).toInteger() > 0
     }
 
+    private void setBreakdownMeasurands(GPathResult viewtag, EventResult result) {
+        viewtag.parent().breakdown?.children()?.forEach { measurand ->
+            if(measurand.name() in Measurand.values().collect{it.getTagInResultXml()}) {
```

Comment:
You actually search for the mesaurand name twice here. I would be sufficient to just
```
Measurand measurand = Measurand.values().find{it.getTagInResultXml() == measurand.name()}
if (measurand) {
```
You can also extract this as a static `byResultXmlTag` method into the Measurand enum, that would make the code even more readable.
Also, you could verify the value first by using `checkIfTagIsThere` (which has an awful name, feel free to rename).